Marlin/src/HAL/STM32/inc/SanityCheck.h

```diff
@@ -51,6 +51,10 @@
   #error "TFT_COLOR_UI, TFT_LVGL_UI and TFT_CLASSIC_UI are currently only supported on STM32H7, STM32F4 and STM32F1 hardware."
 #endif
 
+#if TEMP_SENSOR_BOARD_IS_INTERNAL && defined(ATEMP) && TEMP_BOARD_PIN != ATEMP
```

Comment:
Maybe
```
#if TEMP_SENSOR_BOARD_IS_INTERNAL
  #if !defined(ATEMP) 
    #error "ATEMP is mandatory"
  #else if TEMP_BOARD_PIN != ATEMP
```
because ATEMP is required by code and by message itself

Reply:
The only reason I added that is because in some unknown future `ATEMP` could be defined as pin 0.

Reply:
ok but 0 is still "defined" to test for invalid 0 it should look like:
```#if TEMP_SENSOR_BOARD_IS_INTERNAL && ATEMP && TEMP_BOARD_PIN != ATEMP```